src/python/pants/core/goals/export.py

```diff
@@ -60,7 +62,15 @@ class ExportResult:
     digest: Digest
     # Create these symlinks. Symlinks are created after the digest is materialized,
     # so may reference files/dirs in the digest.
+    # TODO: Remove this functionality entirely? We introduced symlinks as a too-clever way of
+    #  linking from distdir into named caches. However that is risky, so we don't currently use it.
     symlinks: tuple[Symlink, ...]
+    # Run these shell commands as local processes after the digest is materialized.
+    # Note that each string element is an entire command to be parsed and interpreted by the shell.
+    # Each command will be run with the following environment:
+    #  PATH: The pants process's original PATH.
+    #  DIGEST_ROOT: The location under the distdir in which the digest is materialized.
+    post_processing_shell_cmds: tuple[str, ...]
```

Comment:
Aha. I was thinking InteractiveProcess / Workspace would be the tools used here. That seems a bit more disciplined and it doesn't seem like a lot more code. I may be missing snags you hit or thought of though.

Reply:
Hmmm that probably makes sense.

Reply:
The python-specific side still has to pass the process it wants to run as a shell command, but the generic goal side should run it as an InteractiveProcess, yes.

Reply:
Why is that? Passing an InteractiveProcess + (digests + offsets) seems fully generic.

Reply:
Oh, hm, yes, that could work.

Reply:
On second thoughts, I think I'd prefer the current PostProcessingCommand layer, as it emphasizes that:

A) The {digest_root} will be interpolated into the argv
B) The env will have PATH set in it
C) We want to hard-code most fields, e.g., input_digest, restartable, forward_signals_to_process, run_in_workspace and append_only_caches.